akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala

```diff
@@ -485,14 +485,16 @@ object StyleGuideDocExamples {
       @nowarn
       private def counter(remaining: Int): Behavior[Command] = {
         //#pattern-match-without-guard
-        Behaviors.receiveMessage {
-          case Down =>
-            if (remaining == 1) {
-              notifyWhenZero.tell(Done)
-              zero
-            } else
-              counter(remaining - 1)
-        }
+        // `@unchecked` for Scala 3, which doesn't support @nowarn
+        Behaviors.receiveMessage(x =>
+          (x: @unchecked) match {
+            case Down =>
+              if (remaining == 1) {
+                notifyWhenZero.tell(Done)
+                zero
+              } else
+                counter(remaining - 1)
+          })
```

Comment:
The docs where this is pulled in is covering guards where the exhaustiveness check didn't use to work, but does since 2.13.5 so we should probably remove that part of the style guide (not necessarily in this PR though)

Reply:
`@nowarn` is also in the works https://github.com/lampepfl/dotty/pull/12857